From patchwork Tue Sep  6 19:49:04 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968110
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 64E32C38145
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 19:50:42 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 487418000D; Tue,  6 Sep 2022 15:50:27 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 41CAC8000A; Tue,  6 Sep 2022 15:50:27 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 14E7C8000D; Tue,  6 Sep 2022 15:50:27 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0010.hostedemail.com
 [216.40.44.10])
	by kanga.kvack.org (Postfix) with ESMTP id F0A598000A
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 15:50:26 -0400 (EDT)
Received: from smtpin19.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay05.hostedemail.com (Postfix) with ESMTP id D0DB74061A
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:50:26 +0000 (UTC)
X-FDA: 79882702452.19.A8DD863
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf19.hostedemail.com (Postfix) with ESMTP id 4B3D11A005A
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:50:26 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdbZG029984;
	Tue, 6 Sep 2022 19:50:23 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=oKIxAaGaKnywQod4vZUAAWKuSFE1MP+cvJDK2c9ropg=;
 b=p2lJ2HSls7Km4rN7jyDt9noc6tX4Q1Eu89vN3wO1SNPtsTNUCH8o+B5P3dEHe+X1gFZZ
 aIALHL9kxYbB/FjRsgWWGaxYyxy5HA0KB9xo3xWiHb82LEoQrEKya9+PDMBHCq/wvXo5
 kJC8x1CauLFVO09HpzdhzsFh6Bh4aR0buubwKnesehQgsXXZy5lBotUoBELf8iTYJgom
 WbLAtwC434B2lzd0/Rq/pteFS/LieZpAIGFFdpkyEmDhM6E83mHw3QF41AkqpY4sNF2h
 pd0oOUfSqrDyHU68rULx3kQW82IWpxpWwG2jo0JUK2YEhE1T3veZWl2N7Pp1Whw4R9kB 7g==
Received: from phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com
 (phxpaimrmta01.appoci.oracle.com [138.1.114.2])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbyftpvf2-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:50:23 +0000
Received: from pps.filterd
 (phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com [127.0.0.1])
	by phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286H2KIl031383;
	Tue, 6 Sep 2022 19:50:23 GMT
Received: from nam11-co1-obe.outbound.protection.outlook.com
 (mail-co1nam11lp2170.outbound.protection.outlook.com [104.47.56.170])
	by phxpaimrmta01.imrmtpd1.prodappphxaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc3ajns-3
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:50:22 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=FZ9tBrR+Oe/Rbf0J0g0LDrAiW4JmSNukp9VbqCuNkFgusrnkAhgSlLLf6ZzPmN9cvrrf9zOA+mRFAq6fB5vAudQEIojO4CXDi/KglUGLOHgLrhONCE8IV403uN7Hz30ozCQHO2FIWnzlRZd4DNiR2KmqRJo5CEWa2RhEknRw3RRKd6fG88iNdzGxL4cfMHLwh6XYkmLfz3oFPqDb+UlCjl9TuhYnuE6qg10l2qBq6E1fzSXqgWx/elgvb2fkorp8lgkVNjOtE+w6aLa+ewlVhtXnmd3S4nQXWxjWXCrSiN0ioBM0tfli0oSJV7nucqF8HgEygBQ6YOBefJxovzmeuw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=oKIxAaGaKnywQod4vZUAAWKuSFE1MP+cvJDK2c9ropg=;
 b=LKItMA5ZNqeaHFFZdZaIqnPTS+nmpG5SOxR1LiYSY1kXRbefnnYASWLFz8awRbS75gve0wAUqmlyuYBuE4UcWy0horBT9Dw/Hq6yhuiwM3mx7pPlb/sNUt34VenoYJzUp8DZldXJf2SjJgmsArHx5GR7qmmVnFmViGHHn1YkxUs9rAm/Qj8ywOgmXRHDrnU7crccrSuQ4jqQLIdD6gXaITJR3fzm5o+HYgtp33lwOzrkwHcQGwAZme9vsliWVagveuj2VveKHbY0TqLpyH8XRmRYk3B0Gj7o12ZfVV81wwbubAmcdZw8gP+AZXcxw+Shx6zs1z1wS9YYYCaCeGZ95Q==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=oKIxAaGaKnywQod4vZUAAWKuSFE1MP+cvJDK2c9ropg=;
 b=yqfEsTZay1WnzPdYwU3HpU1kCDvb+TbMTPfLn0mc9F44zpFt6a9aRkioDg2T/ehz2qCDdWyrpabi4t7athr3N0q61sJUZmX9DTVBblA6M7cK5eAZ/EkRif7d7bkfUEe472lBbUEa/lgThEfHKW9UjJA1/GAFids5XsfLUNdFPAk=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by CH0PR10MB4876.namprd10.prod.outlook.com (2603:10b6:610:c9::16) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.15; Tue, 6 Sep
 2022 19:50:21 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:50:21 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: "Matthew Wilcox (Oracle)" <willy@infradead.org>,
        Liam Howlett
	<liam.howlett@oracle.com>,
        Vlastimil Babka <vbabka@suse.cz>
Subject: [PATCH v14 62/70] mm/pagewalk: use vma_find() instead of vma linked
 list
Thread-Topic: [PATCH v14 62/70] mm/pagewalk: use vma_find() instead of vma
 linked list
Thread-Index: AQHYwim3D9/zE62xcUu1amnoDHK+bw==
Date: Tue, 6 Sep 2022 19:49:04 +0000
Message-ID: <20220906194824.2110408-63-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: dea0ece6-b113-4cd2-74c3-08da90410823
x-ms-traffictypediagnostic: CH0PR10MB4876:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 mCZvaWqBw8jg5Y0018OF9ZdTB8HIGA4lETeDgI/PrkwLh4j5uvjJA3sNgr/I78FGmTMimKLfT5BhZbwaOSIhOcMl+HBISRY45P4BD06CDJ7KE74m2zqqLXWPMzvNSKrIGQdn5of3gFFsnNmjlmd5lYu3pTwI6SVVQbHdYb41A7CxHe8VQF1UHeP5kcqHAJj+RPwrHlywBF9O0Oj++5e7ssZWyRFMGmvrBazIr1MbKH9d4HTbEuefoZgoFkGaLrzwf0cwhKOoN1V01kGObHALKZDAlr3hZoE47Hh9mbMrEcPQmeTbHu2o/v9dD3Y6yKD9gLPKG0QfhUGhnOEqEVFb6EE4ewhHG8GEV6WSzG1HCRFgQIYaECKx0OYcTfKRd3Rx7ny4XSMfnhwUlItgR9DT+ECUhHFOu0IRM+LgfMrILx0CY7r57eRqyjh2DKzDMB9KT1W01IRPjjgWRRsV8Fse9NQZVyFbrB2UNMC1LkEh6n4U1YSudsXjGop8G/N5MPwIPbLvjv2ijKdQof7RbMJhRq2ad+2rSGsLV0LVWBeZ7DYCth44ObhLlWOMhZjp7qQOTttwaXUOefQRtHmML+6y7dNExAr0ZNBptVbMdwNiBN15CoOyvZacVHCa20gHrfvCbZiOJWXVAAG5lGgmanujdIjAWAI4t5YSMKzDS5vv7chUqB/3GMtKG0AcNpa5Z8uWKUFJSypriIExkTkZM4V7RdHkATC7dnVV4p0IDvtb3blhysmKhxnPOGmic0CoyrIwISoGBGYufUZDGwZbwRJulQ==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(39860400002)(136003)(366004)(376002)(396003)(346002)(8936002)(6486002)(8676002)(4326008)(91956017)(122000001)(66946007)(66446008)(186003)(66476007)(76116006)(64756008)(66556008)(2616005)(38100700002)(38070700005)(478600001)(83380400001)(36756003)(1076003)(5660300002)(4744005)(41300700001)(71200400001)(6666004)(44832011)(26005)(6506007)(54906003)(316002)(86362001)(2906002)(110136005)(6512007);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?Ds5jMVa3ZZ2Dz95Ia503960?=
	=?iso-8859-1?q?VhADzvve4Q1CsEDM9YgHOuE53fsDRmNPFuiyyLos17KaVI2oqsVgtjzOf5ZF?=
	=?iso-8859-1?q?DESG2gPpHuReDivo0YdA4fHK7Qx4PsdklhxcMiIMTqFQ6zIFT52bVhj0v5WP?=
	=?iso-8859-1?q?8iz8ftfR7JlU+bZEWbZ+ADSwnM/Z5IjnqosbIiIJOgvnZUS25saHVUOrnWVr?=
	=?iso-8859-1?q?JiQSnfNOeEsJc99l3Y9pco4fS+hwIeiRdQk3Tu1TULUKvRy9WRPWKkpSJ3Cl?=
	=?iso-8859-1?q?ETuO08OPrCwfMXRurzrDGQF65eE6rDLjaGgODB8Sd4C+fBRIf1Vl8cNyxckb?=
	=?iso-8859-1?q?AiiYD+DvMzsIPq5csBnhoEqbcmKw0MQ5u6qkmgaFOJxBJcMWocMIADSVAUs8?=
	=?iso-8859-1?q?jfvgRn8HZZB2qFp23J43FITQ/ul788jrS7DnoQTt1Yd2YY3pEdW3DkAvWbbl?=
	=?iso-8859-1?q?IHo2y/G1rUcdjMz4ExbJ1BTXhRrldjX/SDe7w0Bx9YfVWnkDthgiDlHFXuTc?=
	=?iso-8859-1?q?V3gTvAHJ+ssH1TxkH9bMSx52+xKeI6c9p+q4AGwZ61xEXQ6yLo0NPvzs99gz?=
	=?iso-8859-1?q?B7l/7lGnhLL5w9zHb+I4hEeJw1Z7cWlVk3j5if0EDb7lY4GUAMFWVm0E4Yrb?=
	=?iso-8859-1?q?Qn6Q5zC2ezzQ03olO9EsWxn9UKmLB7goi/jBF90Q8wikLzmVkiE4Rnhx4nrK?=
	=?iso-8859-1?q?y2xw4E5mtIzX1206cnyfi6BNvw7siLufRro8RDz+rRRdhKJvlc5XOftcUdEs?=
	=?iso-8859-1?q?DZmjFPIZqS9CD9LUpVMmQEqMFq5RXOIZe/lIdq3YAD9fATL7IfSHOLhRjS8u?=
	=?iso-8859-1?q?nFcvpPY7vSNdmYaPc4/tOzfFPx2m4p3pWKVHVZdB8Cdbl4FCnFjkX75sbu9R?=
	=?iso-8859-1?q?gTD9j9zNetqphLQ/AkJq71fmBDlpVAOKZtqctbbRzVNd7TEW0ikzgLgpjbH/?=
	=?iso-8859-1?q?aJ9RkYcXIk6gX7yt/QVyLMQQ1WX4Epq1IE1E55w3SQRufM+BGudl8QYa7naJ?=
	=?iso-8859-1?q?w0CkdDdLsNpAbkHHPJmpnkTsuh0ky3ZMT43h5/hCKq6hPl0J1Qg9Lg1kLCCx?=
	=?iso-8859-1?q?406Orv9tvliNvNE87uUWqjhcVJar7Bv89fUmApro5KRrgcURRYgDzqkBOCbO?=
	=?iso-8859-1?q?Z4URXkJIvC4TkN73WGVgGPjsmSNKEyokAjDa9l1dwW9Fo8v5q+eN5oVpjGZf?=
	=?iso-8859-1?q?dWRmZYP/j6UdluIxsNRC0j3YXAykLmQlW0n7EcWb56x1/LQqIUje8vh6wzhg?=
	=?iso-8859-1?q?3sYxhRlskV2Xi9pS94grCPYFR+W1Fc9aLEzaotL2FsdvUXCWysNOSdOIaVob?=
	=?iso-8859-1?q?xUFybE4ig5sezMny70d/E/vjJh8ROqUdfA00wBAKg0bV/ngHeJ5DYWKU4rHa?=
	=?iso-8859-1?q?7Wt3904ZPu11+EGcljf2Jic7UkFMero9CR5gk+N4MJllrWNcXUD5QmDcYOFH?=
	=?iso-8859-1?q?GxDwjeTPkqTiiaXIFeDQZFMu3EGuuWQ3nActxpoV0PKXcqG6wTF40M8dKWtf?=
	=?iso-8859-1?q?FLcJRJdazISK4FwZcUTtg+PBtlolmaY4tx0KKFUgRXig6otNApMuJdkmyOxe?=
	=?iso-8859-1?q?5irGvTmwHdONjA9fn/pByeII9HgBVgCiK34PKz0mQWkdMhLcdJ0Xz8WK5ZIb?=
	=?iso-8859-1?q?P764GZMuVbhuY2jsBDF79S4ECqLKbofE2E+3UKA=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 dea0ece6-b113-4cd2-74c3-08da90410823
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:49:04.2557
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 OKA+xHy8P87ZWrjlO9oMymlcYe+Btdryihfj/qRWjQQXL2mbwLa+VjKxjYD0r9lSoLpt261Kk2WqcpAel+qHYQ==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: CH0PR10MB4876
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 spamscore=0
 malwarescore=0 mlxscore=0
 suspectscore=0 mlxlogscore=999 phishscore=0 adultscore=0 bulkscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2207270000
 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: CjqNnM_km-N9AkEhNw_Qj-JNHOaf_rZK
X-Proofpoint-GUID: CjqNnM_km-N9AkEhNw_Qj-JNHOaf_rZK
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662493826; a=rsa-sha256;
	cv=pass;
	b=AciFTLrg9Q/wH0Mg7MkLUzLo3YkO9NLnc/hARac8A3RXXWgAJoOPPmjYinrUmw33G4QT5p
	P5cFyyfvqN8Y7Npr1MDVNJMhp8kBSzM+XMGuH1Dr1D5aoaZNAXopPcfOTOyYA9RrRNqR3g
	O4i8NkGKHzx1F7UmVD0/iSHIxmst8So=
ARC-Authentication-Results: i=2;
	imf19.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=p2lJ2HSl;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=yqfEsTZa;
	spf=pass (imf19.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1")
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662493826;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=oKIxAaGaKnywQod4vZUAAWKuSFE1MP+cvJDK2c9ropg=;
	b=nDggu0LaRISZwPAv0AdLS4Vw7oVu1Sbxg3uQ5AHuncbxn8qXLCRWoG7nOjsO2B+o/ytJSP
	LMhZbOPJDWJjwDu8yslUoicSrW/++nhcu9Lje+5fkVbLy5QJ4sakekLgX4x9rCP8oRCL6x
	UZmkvVLL0LnxDwRRe5l9/jIID66rhgE=
X-Rspamd-Queue-Id: 4B3D11A005A
Authentication-Results: imf19.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=p2lJ2HSl;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=yqfEsTZa;
	spf=pass (imf19.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com;
	arc=pass ("microsoft.com:s=arcselector9901:i=1")
X-Rspam-User: 
X-Rspamd-Server: rspam12
X-Stat-Signature: ne3ebd78zw1bjcexhy37qdcya8kte64j
X-HE-Tag: 1662493826-211659
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

walk_page_range() no longer uses the one vma linked list reference.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
---
 mm/pagewalk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/pagewalk.c b/mm/pagewalk.c
index 9b3db11a4d1d..53e5c145fcce 100644
--- a/mm/pagewalk.c
+++ b/mm/pagewalk.c
@@ -456,7 +456,7 @@ int walk_page_range(struct mm_struct *mm, unsigned long start,
 		} else { /* inside vma */
 			walk.vma = vma;
 			next = min(end, vma->vm_end);
-			vma = vma->vm_next;
+			vma = find_vma(mm, vma->vm_end);
 
 			err = walk_page_test(start, next, &walk);
 			if (err > 0) {

